### PR TITLE
Add github.CreateIssue() function

### PR DESCRIPTION
**Pull request by puerco**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:

This PR adds a new function to the github package: `CreateIssue` to file new issues in a GitHub repository.

#### Which issue(s) this PR fixes:
None

#### Special notes for your reviewer:

This function is needed to create the publishing bot issue when we create a new branch on k/k

#### Does this PR introduce a user-facing change?

```release-note
New function `github.CreateIssue` to file new issues in a GitHub repository
```

### CODE REVIEW

File: pkg/github/github.go [hunk 1]
```diff
@@ -128,6 +132,15 @@ type Client interface {
 	) (*github.IssueComment, *github.Response, error)
 }
 
+// NewIssueOptions is a struct of optional fields for new issues
+type NewIssueOptions struct {
+	Assignee  string   // GitHub handle of main assignee
```
Comment (xmudrii) [inline pkg/github/github.go:137]: Why do we need to distinguish between the main assignee and other assignees? Is that coming from the GitHub API?

File: pkg/github/github.go [hunk 2]
```diff
@@ -333,6 +346,19 @@ func (g *githubClient) CreatePullRequest(
 	return pr, nil
 }
 
+func (g *githubClient) CreateIssue(
+	ctx context.Context, owner, repo string, req *github.IssueRequest,
+) (*github.Issue, error) {
+	// Create the issue on github
+	issue, _, err := g.Issues.Create(ctx, owner, repo, req)
+	if err != nil {
+		return issue, errors.Wrap(err, "creating pull request")
```
Comment (xmudrii) [inline pkg/github/github.go:355]: Nit:
```suggestion
		return issue, errors.Wrap(err, "creating issue")
```

File: pkg/github/github.go [hunk 3]
```diff
@@ -686,6 +712,37 @@ func (g *GitHub) UploadReleaseAsset(
...
+	if nio.State == "open" || nio.State == "closed" || nio.State == "all" {
+		request.State = &nio.State
+	}
+
+	if len(nio.Labels) > 0 {
+		request.Labels = &nio.Labels
+	}
+
+	if len(nio.Assignees) > 0 {
+		request.Labels = &nio.Assignees
```
Comment (xmudrii) [inline pkg/github/github.go:728]: This seems wrong. Should we really assign assignees to labels?

File: pkg/github/github.go [hunk 4]
```diff
@@ -686,6 +712,37 @@ func (g *GitHub) UploadReleaseAsset(
 	return asset, nil
 }
 
+// ToRequest builds an issue request from the set of options
+func (nio *NewIssueOptions) toRequest() *github.IssueRequest {
```
Comment (xmudrii) [inline pkg/github/github.go:716]: I don't see that we use `Milestone` anywhere. Is that as expected?

**Review by xmudrii — COMMENTED** [review]
Overall looks okay, left a few comments.

File: pkg/github/github.go (hunk 4 above)
Comment (puerco) [reply pkg/github/github.go:715]: Milestones need an additional function to resolve their title to the numeric ID. Currently is ignored, it is coming in a followup PR.

File: pkg/github/github.go (hunk 3 above)
Comment (puerco) [reply pkg/github/github.go]: Good catch. Fixed :+1:

File: pkg/github/github.go (hunk 1 above)
Comment (puerco) [reply pkg/github/github.go]: Yes it comes from the GitHub API. But you gave me an idea, we can simplify our options struct more and assign them automatically from a single slice of usernames :+1:

File: pkg/github/github.go (hunk 4 above)
Comment (puerco) [reply pkg/github/github.go:715]: `github.GetMilestone()` followup PR:  https://github.com/kubernetes/release/pull/1965

**Review by cpanato — APPROVED** [review]
/lgtm
/hold for @xmudrii

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve
/hold cancel

Great job, @puerco!

File: pkg/github/github.go [hunk 5]
```diff
@@ -686,6 +711,39 @@ func (g *GitHub) UploadReleaseAsset(
...
+	request := &github.IssueRequest{}
+
+	if nio.State == "open" || nio.State == "closed" || nio.State == "all" {
+		request.State = &nio.State
+	}
```
Comment (saschagrunert) [inline pkg/github/github.go:720]: We could error in the case the state is invalid.

File: pkg/github/github.go [hunk 6]
```diff
@@ -128,6 +132,14 @@ type Client interface {
 	) (*github.IssueComment, *github.Response, error)
 }
 
+// NewIssueOptions is a struct of optional fields for new issues
+type NewIssueOptions struct {
+	Milestone string   // Name of milestone to set
+	State     string   // open, closed or all. Defaults to "open"
```
Comment (saschagrunert) [inline pkg/github/github.go:138]: I'm wondering why we have this field. What is the use case behind creating an issue with the state `closed` or `all`?

Comment (puerco) [reply pkg/github/github.go:138]: :100:  I was just porting it without thinking about the application. I'll drop it.

**Pull request merged**